cmd/README.md

```diff
@@ -0,0 +1,618 @@
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+Finally we can modify our `setGreeting()` function 
+
+```
+func setGreeting(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SetGreetingGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// attempts to unpack [input] into the arguments to the SetGreetingInput.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStr, err := UnpackSetGreetingInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+	// setGreeting is the execution function of "SetGreeting(name string)" and sets the recipient in the string returned by hello world
+	res := common.LeftPadBytes([]byte(inputStr), common.HashLength)
+	accessibleState.GetStateDB().SetState(IHelloWorldAddress, common.BytesToHash([]byte("recipient")), common.BytesToHash(res))
+
+	// this function does not return an output, leave this one as is
+	packedOutput := []byte{}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+## Step 4 
+
+Let's now modify `params/precompile_config.go`. We can `CTRL F` for `ADD YOUR PRECOMPILE HERE`. 
+
+Let's add our key. 
+
+![](2022-09-01-23-20-53.png)
+
+Let's add our precompile config to `PrecompileUpgrade`
+
+![](2022-09-01-23-19-42.png)
+
+Finally, we can add a getter function. 
+
+![](2022-09-01-23-23-20.png)
+
+Done! All we had to do was follow the comments.
+
+
+## Step 5 
+
+Let's add our precompile upgrade in `params/config.go`. We can `CTRL F` for `ADD YOUR PRECOMPILE HERE`. 
+
+Let's add the bool to check if our precompile is enabled.  
+
+![](2022-09-01-23-35-47.png)
+
+We can now add it to the Avalanche rules. 
+
+![](2022-09-01-23-36-56.png)
+
+Lastly, we can add `IsHelloWorld` which checks if we are equal or greater than the fork `blockTimestamp`. 
+We use this to see if we should enable the precompile. 
+
+![](2022-09-01-23-32-51.png)
+
+## Step 6
+
+Add your solidity interface and test contract to `contract-examples/contracts`
+
+We already have our interface in  `contract-examples/contracts`. 
+Let's add our contract.
+
+```
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IHelloWorld.sol";
+
+// HelloWorld shows how the HelloWorld precompile can be used in a smart conract
+contract HelloWorld {
+  address constant HELLO_WORLD_ADDRESS = 0x0200000000000000000000000000000000000004;
+  HelloWorld helloWorld = HelloWorld(HELLO_WORLD_ADDRESS);
+
+  function sayHello() public returns (string memory) {
```

Comment:
this might be very confusing for the reader. they don't need to add the same function signatures as the precompile. we can rename `sayHello` in the precompile to something like `getGreeting` and use this `sayHello` function like:
```
  function sayHello() public returns (string memory) {
   return helloWorld.getGreeting();
  }
``` 
or change the `sayHello` in this contract to `getHello` etc.
This should be to demonstrate that we can extend the behaviour in precompile with these wrapping contracts.

Reply:
Thanks added a note about that